src/tests.rs

```diff
@@ -0,0 +1,284 @@
+use crate::benchmark::Benchmark;
+use crate::blending::{
+    blend_images, get_blending_algorithm, is_algorithm_multiplied, multiply_image, BlendAlgorithm,
+};
+use crate::errors::PConvertError;
+use crate::parallelism::{ResultMessage, ThreadPool};
+use crate::utils::{read_png_from_file, write_png_parallel, write_png_to_file};
+use image::codecs::png::{CompressionType, FilterType};
+use image::Rgba;
+use std::fmt;
+use std::fmt::{Display, Formatter};
+
+#[cfg(test)]
+use crate::constants;
+#[cfg(test)]
+use image::ImageFormat;
+#[cfg(test)]
+use std::str::FromStr;
+
+#[test]
+fn test_compose() {
+    let test_dir = "assets/test/";
+    let mut benchmark = Benchmark::new();
+    let backgrounds = vec![
+        Background::Alpha,
+        Background::Blue,
+        Background::Texture,
+        Background::White,
+    ];
+
+    // composes with different combinations of blending algorithms and backgrounds
+    for background in backgrounds {
+        for algorithm in constants::ALGORITHMS.iter() {
+            compose(
+                &test_dir,
+                BlendAlgorithm::from_str(algorithm).unwrap(),
+                background.clone(),
+                CompressionType::Fast,
+                FilterType::NoFilter,
+                &mut benchmark,
+            )
+            .expect(&format!(
+                "failed composing with algorithm={} background={} compression=Fast filter=NoFilter",
+                algorithm, background
+            ));
+        }
+    }
+}
+
+#[test]
+fn test_compose_parallel() {
+    let test_dir = "assets/test/";
+    let mut benchmark = Benchmark::new();
+    let backgrounds = vec![
+        Background::Alpha,
+        Background::Blue,
+        Background::Texture,
+        Background::White,
+    ];
+
+    // composes with different combinations of blending algorithms and backgrounds
+    for background in backgrounds {
+        for algorithm in constants::ALGORITHMS.iter() {
+            compose_parallel(
+                &test_dir,
+                BlendAlgorithm::from_str(algorithm).unwrap(),
+                background.clone(),
+                CompressionType::Fast,
+                FilterType::NoFilter,
+                &mut benchmark,
+            )
+            .expect(&format!(
+                "failed composing with algorithm={} background={} compression=Fast filter=NoFilter",
+                algorithm, background
+            ));
+        }
+    }
+}
+
+#[test]
+fn test_convert() {
+    let test_dir = "assets/test/";
+    let test_file = "tux.png";
+    let test_file_out = "result_tux.png";
```

Comment:
`let`s are by definition  constant values in rust. They can only be mutated with `let mut`.
`const` is used only for globals

Reply:
> lets are by definition constant values in rust. They can only be mutated with let mut.

I know, but the fact that the behavior is the same doesn't change my point.

> const is used only for globals

Where did you get that idea from? You can't have global `let`s but you can have scoped `const`s. In any case, this is the same as the `5` that I commented somewhere else: the values should be located at the top of the file, uppercased, etc. Loosely related: https://github.com/ripe-tech/ripe-building/blob/master/src/ripe_building/runners/verify.py#L80

Reply:
Ok, I really thought it had to be global values, so far I've only seen Rust code with const and global constants. But it does make sense since constants end up being global. So it does make sense you can have scoped consts. I will however make them global here, similar to the reference you gave